File: BAEKJOON/data_structure/javascript/9012.js
const fs = require("fs");
let input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().trim()
    : fs
        .readFileSync(__dirname + "/input.txt")
        .toString()
        .trim();

input = input.split("\n");
input.shift();

function checker(ps) {
  const stack = [];

  for (let p of ps) {
    if (p === "(") {
      stack.push(p);
    } else if (stack.length === 0) {
      return "NO";
    } else {
      stack.pop();
    }
  }

  if (stack.length === 0) {
    return "YES";
  } else {
    return "NO";
  }
}

const result = [];
for (let ps of input) {
  result.push(checker(ps));
}

console.log(result.join("\n"));

File: BAEKJOON/data_structure/python/9012.py
""" 
22.07.05
BOJ : 9012 / 자료구조
"""
import sys

n = int(input())
for _ in range(n):
    string = list(sys.stdin.readline().strip())
    init_stack = []

    for i in string:
        if i == '(':
            init_stack.append(i)
        else:
            if not init_stack:
                init_stack.append(i)
            else:
                if init_stack[-1] == '(':
                    init_stack.pop()
                else:
                    init_stack.append(i)
    
    if not init_stack:
        print('YES')
    else:
        print('NO')
